### PR TITLE
CORTX-30781 Add regression test for default vs other namespace

### PR DESCRIPTION
**Pull request by walterlopatka**
## Description
This change adds a feature to the regression test `test_deploy.py` to specify a namespace.
When namespace is specified it overrides the namespace specified in the cluster config file.

In addition, a new test that specifies "default" namespace is added (to ensure that deployment
into the default namespace works as expected).

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds new functionality)
- [ ] Breaking change (bug fix or new feature that breaks existing functionality)
- [ ] Third-party dependency update
- [ ] Documentation additions or improvements
- [x] Code quality improvements to existing code or test additions/updates

## Applicable issues
- This change is related to an issue: CORTX-30781

## How was this tested?
I have run this via ./testrunner.py -c <config> -t testlists/deploy_all.py

## Checklist

- [x] The change is tested and works locally.
- [ ] New or changed settings in the solution YAML are documented clearly in the README.md file.
- [x] All commits are signed off and are in agreement with the [CORTX Community DCO and CLA policy](https://github.com/Seagate/cortx/blob/main/doc/dco_cla.md).

If this change addresses a CORTX Jira issue:

- [x] The title of the PR starts with the issue ID (e.g. `CORTX-XXXXX:`)

### CODE REVIEW

File: test/regression/test_deploy.py [hunk 1]
```diff
@@ -1,13 +1,46 @@
...
 
+import yaml
+
 from checker import Checker
 from cluster import Cluster
 from utils import Logger, StopWatch
 
 
+def verify_pods_in_namespace(checker, namespace):
+    stdout = subprocess.Popen(['kubectl', 'get', 'pods', '-A'], # nosec
```
Comment (osowski) [inline test/regression/test_deploy.py:16]: This function is `verify_pods_in_namespace`, but then the command you use to get pods is using the `-A` flag which gets all pods across all namespaces. Should this command be updated to use `-n namespace` instead?

**Review by osowski — COMMENTED** [review]
Only one comment on usage of `kubectl get pods -A`, but other than that looks good to me.

File: test/regression/test_deploy.py (hunk 1 above)
Comment (walterlopatka) [reply test/regression/test_deploy.py:16]: Good point.... not sure why I would have done it this way.  🤔

Comment (walterlopatka) [reply test/regression/test_deploy.py]: Ok, I fixed it.

**Review by keithpine — APPROVED** [review]

**Review by osowski — APPROVED** [review]
lgtm

**Pull request merged**